protos/google/cloud/compute/node_groups/v1/node_groups.proto

```diff
@@ -656,6 +669,44 @@ message PatchNodeGroupRequest {
   ];
 }
 
+// Request message for PerformMaintenance.
+message PerformMaintenanceRequest {
+  // Name of the node group scoping this request.
+  string node_group = 1
+      [(google.api.field_behavior) = REQUIRED, json_name = "nodeGroup"];
+
+  // The NodeGroupsPerformMaintenanceRequest for this request.
+  optional google.cloud.cpp.compute.v1.NodeGroupsPerformMaintenanceRequest
+      node_groups_perform_maintenance_request_resource = 2
+      [json_name = "__json_request_body"];
+
+  // Project ID for this request.
+  string project = 3 [
+    (google.api.field_behavior) = REQUIRED,
+    (google.cloud.operation_request_field) = "project",
+    json_name = "project"
+  ];
+
+  // An optional request ID to identify requests. Specify a unique request ID so
+  // that if you must retry your request, the server will know to ignore the
+  // request if it has already been completed. For example, consider a situation
+  // where you make an initial request and the request times out. If you make
+  // the request again with the same request ID, the server can check if
+  // original operation with the same request ID was received, and if so, will
+  // ignore the second request. This prevents clients from accidentally creating
+  // duplicate commitments. The request ID must be a valid UUID with the
+  // exception that zero UUID is not supported (
+  // 00000000-0000-0000-0000-000000000000).
+  optional string request_id = 4 [json_name = "requestId"];
```

Comment:
Just an observation: we could automatically generate these UUIDs if the protos had the right annotations.